packages/dds/merge-tree/src/client.ts

```diff
@@ -712,6 +712,32 @@ export class Client {
         return this._mergeTree.getPosition(segment, currentSeq, clientId, localSeq);
     }
 
+    /**
+     * Rebases a (local) position from the perspective `{ seq: seqNumberFrom, localSeq }` to the perspective
+     * of the current sequence number. This is desirable when rebasing operations for reconnection.
+     *
+     * If the position refers to a segment/offset that was removed by some operation between `seqNumberFrom` and
+     * the current sequence number, returns undefined.
+     */
+    public rebasePositionWithoutSegmentSlide(
```

Comment:
could you file some work under "SharedString quality improvements" feature to clean up this method + `rebasePosition`? There's definitely a more holistic set of APIs we could expose which provide ways to resolve positions at given perspectives into segment+offset and vice versa. An immediate improvement to the current scenario IMO is to expose findReconnectionPosition, remove the current rebasePosition and replace call sites with this method, moving segment slide responsibility to call sites. I suspect that even these two APIs can be combined concisely into some existing methods though.

Reply:
Created [2381](https://dev.azure.com/fluidframework/internal/_workitems/edit/2381) for this item.